src/ln/channelmonitor.rs

```diff
@@ -911,7 +915,23 @@ impl ChannelMonitor {
 					htlc_idxs.push(None);
 					values.push(outp.value);
 					total_value += outp.value;
-					break; // There can only be one of these
+				} else if outp.script_pubkey.is_v0_p2wpkh() {
+					match self.key_storage {
+						KeyStorage::PrivMode { revocation_base_key: _, htlc_base_key: _, delayed_payment_base_key: _, ref payment_base_key, prev_latest_per_commitment_point: _, latest_per_commitment_point: _ } => {
+							let per_commitment_point = PublicKey::from_secret_key(&self.secp_ctx, &per_commitment_key);
+							if let Ok(local_key) = chan_utils::derive_private_key(&self.secp_ctx, &per_commitment_point, &payment_base_key) {
+								spendable_outputs.push(SpendableOutputDescriptor::DynamicOutput {
+									outpoint: BitcoinOutPoint { txid: commitment_txid, vout: idx as u32 },
+									key: local_key,
+									witness_script: None,
+									to_self_delay: 0,
+								});
+							}
+						}
+						KeyStorage::SigsMode { .. } => {
+							unimplemented!();
```

Comment:
I dont think this is really unimplemented, but rather we're not going to implement this - instead it should just be a TODO noting that we need to ensure an offline client will generate the event when it comes back online after only the watchtower saw the transaction.